internal/chain/chain.go

```diff
@@ -0,0 +1,83 @@
+package chain
```

Comment:
Something makes me a bit wary of having `internal/chain` and `internal/rpc` and `internal/utils` packages. I suppose it isn't obvious to me what they are for. Could we combine them all intro the `internal/nitro-network` package (could still keep the source files separate, but all in one folder)?

Reply:
I initially started with the approach of one `internal/nitro-network` package. I then realized that I was pulling needed utilities out of other parts of our codebase and aggregating those in the `internal/nitro-network` package. So the package became a disorganized, miscellaneous directory of utilities.

My current philosophy is:
- If a utility is useful for consumers of `go-nitro`, the utility should be placed in a relevant package and exported. This is our usual pattern.
- If a utility is used only internally by a single package, the utility should be placed in that package and not exported. This is also our usual pattern.
- If a internal utility is used by multiple packages, it should be placed in a narrowly scoped internal package.

More than happy to discuss and reconsider this philosophy.

Reply:
This sounds good to me 👍